File: Cenfomon/Capa_Logica/src/main/java/com/cenfomon/Cenfomon_Capa_Logica/bl/entities/Movimiento.java
package com.cenfomon.Cenfomon_Capa_Logica.bl.entities;

public class Movimiento {
    /**
     * Atributos
     */
    int id;
    String name;
    short puntos;
    ETipoMovimiento tipo;
    TipoElemento tipoElemento;
    Object potenciador;
    /**
     * get and set
     */
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public short getPuntos() {
        return puntos;
    }

    public void setPuntos(short puntos) {
        this.puntos = puntos;
    }

    public ETipoMovimiento getTipo() {
        return tipo;
    }

    public void setTipo(ETipoMovimiento tipo) {
        this.tipo = tipo;
    }

    public TipoElemento getTipoElemento() {
        return tipoElemento;
    }

    public void setTipoElemento(TipoElemento tipoElemento) {
        this.tipoElemento = tipoElemento;
    }

    public Object getPotenciador() {
        return potenciador;
    }

    public void setPotenciador(Object potenciador) {
        this.potenciador = potenciador;
    }
}

File: Cenfomon/Cenfomon_COI/bl/f_movimiento/Producto/Movimiento.ts
import {ITipoElemento} from "../../fa_TipoElemento/productoAbstracto/ITipoElemento";
import {Helper} from "../../factoryMethodItems/Helper";

export abstract class Movimiento{
    private _id :number;
    private _name : string;
    private _puntos: number;
    private _tipo :ITipoElemento;
    /**
     * Constructor
     */


    constructor() {
        this._id = new Helper().generarID();
        this._name = "";
        this._puntos = 0;
        this._tipo = new class implements ITipoElemento {};
    }

    /**
     * Get y sets
     */


    getid(): number {
        return this._id;
    }

    setid(value: number): void {
        this._id = value;
    }

    getname(): string {
        return this._name;
    }

    setname(value: string): void {
        this._name = value;
    }

    get puntos(): number {
        return this._puntos;
    }

    set puntos(value: number) {
        this._puntos = value;
    }

    get tipo(): ITipoElemento {
        return this._tipo;
    }

    set tipo(value: ITipoElemento) {
        this._tipo = value;
    }

}